SI/Sem.04/Pract.04/Task11.cpp

```diff
@@ -1,30 +1,87 @@
-#include <iostream>
+// Task11.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
 
+#include <iostream>
 using namespace std;
 
-int biggestCommonFactor(int a, int b) {
-	int result = 1;
-	int smaller = (a > b) ? b : a;
-	for (int i = 2; i <= smaller; i++) {
-		if (a % i == 0 && b % i == 0) {
-			result = i;
+//int smallestDivisor(int a, int b, int c, int d)
+//{
+//	int smallestNum;
+//
+//	if (a < b && a < c && a < d)
+//	{
+//		smallestNum = a;
+//	} else if (b < a && b < c && b < d)
+//	{
+//		smallestNum = b;
+//	} else if (c < a && c < b && c < d)
+//	{
+//		smallestNum = c;
+//	}
+//	else
+//	{
+//		smallestNum = d;
+//	}
+//	
+//}
+
+int smallestDivisor(int a, int b)
+{
+	if (a < b)
+	{
+		int temp = a;
+		a = b;
+		b = temp;
+	}
+
+	int divisor;
+	while (true)
+	{
+		if (a % b == 0)
+		{
+			divisor = b;
+			return divisor;
 		}
+
+		divisor = a % b;
+		a = b;
+		b = divisor;
+	}
+}
+
+int smallestDivisorForFourNums(int a, int b, int c, int d)
+{
+	int firstDiv = smallestDivisor(a, b);
+	int secondDiv = smallestDivisor(c, d);
+
+	if (firstDiv > secondDiv)
+	{
+		int temp = firstDiv;
+		secondDiv = firstDiv;
+		firstDiv = temp;
 	}
 
-	return result;
+	int finalDiv;
+	for (int i = secondDiv; i >= firstDiv; --i)
+	{
+		if (a % i == 0 && b % i == 0 && c % i == 0 && d % i == 0)
+		{
+			return i;
+		}
+	}
```

Comment:
Easier way is just to find the gcd of a and b, then find the gcd of c and d, finally find the gcd of the two results